include/exadg/fluid_structure_interaction/acceleration_schemes/parameters.h

```diff
@@ -34,6 +34,15 @@ enum class AccelerationMethod
   IQN_IMVLS
 };
 
+enum class UpdateMethod
+{
+  Undefined,
+  Implicit,
+  GeometricExplicit,
+  ImplicitPressureStructure,
+  ImplicitVelocityStructure
+};
```

Comment:
I think I would not implement it like this, but instead use three boolean parameters:

```C++
bool update_ale;
bool update_fluid_velocity;
bool update_fluid_pressure;
```
Default values are `true` such that the standard Dirichlet-Neumann-scheme with all couplings is the default.

Advantages:
- intuitive to understand for a user in my opinion
- the notion of implicit and explicit in the above `enum` is not clear a priori in my opinion without documentation. Moreover, the terms `ImplicitPressureStructure` and `ImplicitVelocityStructure` do not intuitively communicate that they inherently involve `GeometricExplicit` in my opinion.
- the above `enum` does not include the variant `GeometricImplicit_PressureStructure` or `GeometricImplicit_VelocityStructure`. This example also shows why I do not like the `Implicit`-wording in the above enum. One might in fact expect `GeometricImplicit` for the last two variants because of `Implicit` in the name, but this is not the case!?
- The above enum requires an if-else if-implementation with currently three branches. Realizing also the two additional variants mentioned here would require one/two more branche(s) following the current implementation style.
- using 3 booleans instead would be very close to the actual implementation of the FSI coupling scheme and the fluid single field solver with exactly these parameters in the interface. 
- The implementation in the FSI code for all 6 variants would collaps to a single "branch"
```C++
if(iter == 0 or update_ale)
  solve_subproblem_ale(displacement_structure);
solve_subproblem_fluid(iteration,  iter==0 or update_velocity, iter== 0 or update_pressure);
solve_subproblem_structure(iteration);
```

for all 6 variants with the 3 boolean parameters (principally 8 variants, but `update_velocity = update_pressure = false` is excluded since this is equivalent to no FSI-coupling at all).

Reply:
Good point, this can easily be way more intuitive and more general, i.e., covering more combinations.
 I would want to name them a bit differently though, since with the current name, one might think they are otherwise not updated at all. But I can't find a different name; let's keep the ones you suggested! (or any ideas?)